app/assets/javascripts/discourse/app/models/group.js

```diff
@@ -293,6 +294,10 @@ const Group = RestModel.extend({
     return ajax(`/groups/${this.id}`, {
       type: "PUT",
       data: { group: this.asJSON() }
+    }).then(data => {
+      if (data.route_to) {
+        DiscourseURL.routeTo(data.route_to);
+      }
```

Comment:
The API will return a value back in the JSON response if it wants the client side app to do a redirect... Is there a better way to do this?

Reply:
To answer myself: the API probably should just return an error, and let the client work out where to redirect to.

Reply:
This is a bit odd, mainly because the model itself shouldn't be responsible for the redirect. The component or controller that calls `save()` should do this in the `then()` method on the promise.